KR plop setup

kjreczek opened the pull request:
I installed the plop library and created file templates for the base component, it works like this: in the terminal we go to the folder where we want to create a new folder with this component and run plop with the command: _npm run plop_
Then we give the component name and that's it. Templates for this are in a special plop-templates folder, while the configuration is in plopfile.js. I did not create anything complicated because I do not really know what our future components will look like and if necessary, we can make such templates ourselves in the coming weeks.

I recommend testing to see how it works. I branched out of Robert's branch.

fszwedo commented:
I went to /frontend/src/app and run the command there but the files were generated in the root 'frontend' folder anyway - I think the components should be created somewhere deeper in the structure right?

fszwedo commented:
and if we're going for redux then maybe a similar flow for actions/reducers and stuff would be useful

robert-czwartosz commented on frontend/package.json:
```diff
@@ -0,0 +1,45 @@
+  "name": "frontend",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@reduxjs/toolkit": "^1.5.0",
+    "@testing-library/jest-dom": "^4.2.4",
+    "@testing-library/react": "^9.5.0",
+    "@testing-library/user-event": "^7.2.1",
+    "@types/jest": "^24.9.1",
+    "@types/node": "^12.20.6",
+    "@types/react": "^16.14.5",
+    "@types/react-dom": "^16.9.11",
+    "@types/react-redux": "^7.1.16",
+    "plop": "^2.7.4",
```
I think plop should be in devDependencies

kjreczek replied:
I was inspired that when I merged your branch, there were not devDependencies. I will change that.

kjreczek commented:
@fszwedo 
I did not notice that it adds to the wrong folder. 
I changed the plop configuration and when you run plop, it also asks question about destination path of the components files.

kjreczek commented:
I changed the plop configuration so that it suggest the default path if nothing is pass. I added a template with a Slice file from Robert's example and changed template .css to module.css.

kjreczek commented:
I merged Justyna's branch and added template stories to plop. @fszwedo @robert-czwartosz

fszwedo reviewed (approved):
all ok for me :)
